CMakeLists.txt

```diff
@@ -91,6 +91,8 @@ target_compile_features(concurrencpp PUBLIC cxx_std_20)
 set_target_properties(concurrencpp PROPERTIES
         CXX_VISIBILITY_PRESET hidden
         VISIBILITY_INLINES_HIDDEN ON
+        VERSION "${PROJECT_VERSION}"
+        SOVERSION "${PROJECT_VERSION_MAJOR}"
```

Comment:
@David-Haim please be aware of this. The ABI version is also now the project's major version component.

Reply:
With these two properties and assuming `project(... VERSION 0.1.5)`, the installation on .so platforms will look like this:

```
namelink:     libconcurrencpp.so       -> libconcurrencpp.so.0.1.5 (symlink)
soname:       libconcurrencpp.so.0     -> libconcurrencpp.so.0.1.5 (symlink)
real library: libconcurrencpp.so.0.1.5
```

Note that the `0` in the soname may be disjoint from the version embedded in the real library's name, because it (sort of) means ABI version. I just wanted you to be aware of this.

Reply:
We can always add another variable for abi version. Assuming you are using semantic versioning, then 0 should be fine in that it is indicating it is a pre-release version, so there is no stability guarantees.